app/src/main/java/com/example/firstapp/EmailValidator.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.firstapp
+
+import java.util.regex.Pattern
+
+object EmailValidator {
+
+    private const val PATTERN = "(?:[a-z0-9!#\$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#\$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"
+
+    fun isValidEmail(email: String?): Boolean {
+        return email?.let {
+            Pattern.matches(PATTERN, it)
```

Comment:
if your pattern returns false when the email is empty, you could simplify this method to 
`fun isValidEmail(email: String?) = Pattern.matches(PATTERN, email.orEmpty)`